csharp-basics/exercises/Tests/ScooterRentalService/ScooterRentalService/RentalCompany.cs

```diff
@@ -0,0 +1,159 @@
+﻿namespace ScooterRentalService
+{
+    public class RentalCompany : IRentalCompany
+    {
+        private readonly IScooterService _scooterService;
+        private readonly Dictionary<string, DateTime> _rentalStartTime = new Dictionary<string, DateTime>();
+        private readonly Dictionary<int, decimal> _yearlyIncome = new Dictionary<int, decimal>();
+        private readonly ITimeProvider _timeProvider;
+        private const int SomeMinimumYear = 2023;
+        private const decimal MAX_DAILY_CHARGE = 20M;
+        private const int MINUTES_IN_DAY = 1440;
+
+        public string Name { get; private set; }
+
+        public RentalCompany(string companyName, IScooterService scooterService, ITimeProvider timeProvider)
+        {
+            _scooterService = scooterService ?? throw new ArgumentNullException(nameof(scooterService));
+            _timeProvider = timeProvider ?? throw new ArgumentNullException(nameof(timeProvider));
+            Name = companyName ?? throw new ArgumentNullException(nameof(companyName));
+        }
+
+        public void StartRent(string id)
+        {
+            var scooter = _scooterService.GetScooterById(id);
+            if (scooter == null)
+            {
+                throw new InvalidOperationException($"Scooter with ID: {id} does not exist.");
+            }
+            if (_rentalStartTime.ContainsKey(id))
+            {
+                throw new InvalidOperationException($"Scooter with ID: {id} is already being rented.");
+            }
+            _rentalStartTime[id] = _timeProvider.Now;
+        }
+
+        public decimal EndRent(string id)
```

Comment:
vajadzētu sadalīt pa metodēm. :) ļoti gara metode.. nav pārskatāmi un jāiedziļinās lai saprastu. :)